src/pages/upstreams/TargetList.vue

```diff
@@ -34,10 +40,55 @@ defineOptions({
 const route = useRoute()
 const toaster = useToaster()
 const { t } = useI18n()
+const infoStore = useInfoStore()
+
+const { isHybridMode } = storeToRefs(infoStore)
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const upstream = ref<any>()
 
 const upstreamId = computed(() => (route.params.id as string) ?? '')
 const cacheIdentifier = computed(() => `targets-${upstreamId.value}`)
 
+/**
+ * Check whether this target's parent upstream is using healthchecks.
+ *
+ * Conditions here refer to:
+ * https://github.com/Kong/kong-ee/blob/21cd7e3b609717d82302684ad3e75582707754c4/kong/runloop/balancer/healthcheckers.lua#L233-L243
+ *
+ * @returns {boolean}
+ * - `true` if using healthchecks or unknown (insufficient permission to the upstream)
+ * - `false` if not using
+ */
+const isUpstreamUsingHealthchecks = computed(() => {
+  if (upstream.value === undefined) {
+    // Returning `true` here because we do not know if the parent upstream is using
+    // healthchecks due to insufficient permission
+    return true
+  }
+
+  const active = upstream.value.healthchecks?.active
+  const passive = upstream.value.healthchecks?.passive
+
+  return (
+    (active?.healthy?.interval ?? 0) !== 0 ||
+    (active?.unhealthy?.interval ?? 0) !== 0 ||
+    (passive?.unhealthy?.tcp_failures ?? 0) !== 0 ||
+    (passive?.unhealthy?.timeouts ?? 0) !== 0 ||
+    (passive?.unhealthy?.http_failures ?? 0) !== 0
+  )
+})
+
+onBeforeMount(async () => {
+  try {
+    const response: AxiosResponse = await apiService.get(`upstreams/${upstreamId.value}`)
```

Comment:
nit: we can use a more specific method:
```suggestion
    const response: AxiosResponse = await apiService.findRecord('upstreams', upstreamId.value)
```